granulate_utils/linux/devkmsg.py

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+import errno
+import os
+import time
+from typing import List, Tuple
+
+# See linux/printk.h
+CONSOLE_EXT_LOG_MAX = 8192
+
+
+class DevKmsgProvider:
```

Comment:
What about the `Provider` interface? I suppose that can be moved into this repo as well - since we'll want to support the `EmptyProvider` for old kernels / low privileges cases. (and at some point we'll add support for kernels < 3.5 so it should also share the same interface)